KAN-82-블로그페이지레이아웃제작

kimgorok opened the pull request:
## 😀 제목 : [블로그] 레이아웃 제작

<br/>

## 🔎 작업 내용



- 어떤 변경 사항이 있었나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [x] 주석 추가 및 수정
- [ ] 문서 수정
- [x] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [x] 파일 혹은 폴더명 수정
- [x] 파일 혹은 폴더 삭제

<br/>

## 🎨 이미지 첨부

![image](https://github.com/user-attachments/assets/a1726bd1-0161-45aa-9961-704ab8dcfb78)

![image](https://github.com/user-attachments/assets/3eff8398-3fb2-493c-878e-06e0581fa879)

![image](https://github.com/user-attachments/assets/d57d5e01-06b8-4c55-84d7-114949004ce7)

![image](https://github.com/user-attachments/assets/efd2f860-53f7-4210-8d63-d15648c8fd19)


<br/>

## 🔧 앞으로의 과제

- 다음으로 할 일은 무엇인가요??

1. 블로그 페이지의 폴더구조가 가독성을 해치지는 않을지, 개선점은 있을지 조금 더 생각해보겠습니다.
2. 더 많아질 컴포넌트들을 어떻게 관리할지 생각해보겠습니다.
3. api가 연결이 되면 본인의 블로그(포스트, 댓글)인지 확인하는 로직을 추가합니다.
4. 코드 리팩토링을 진행합니다.
5. 댓글 아이콘을 추가합니다.
6. 상수 관리를 진행합니다.

  <br/>

## ➕ 추가 코멘트

- (10.01) 
1. /blog/[userId] 페이지를 세팅했습니다.
2. 배경 색을 바꾸고 범용 스타일을 적용했습니다.
3. Icons.tsx를 수정하고 새로운 아이콘을 추가했습니다.
4. 최근 포스트 목록 컴포넌트를 세팅하고 테스트 코드를 추가했습니다.
5. /blog/[userId]/[postTitle] 페이지를 세팅했습니다.
6. 포스트 상세 조회와 관련한 컴포넌트를 생성했습니다.
7. 뒤로가기 아이콘을 생성하고 적용했습니다.

- (10.02)
1. /blog 하위의 layout.tsx를 작성했습니다.

<br/>

kimgorok committed:
🎉init: [블로그] KAN-82 블로그 페이지 세팅

- /blog/[userId] 라우팅
- UserProfile컴포넌트 폴더 이동

/blog/[userId]

kimgorok committed:
💄design: [블로그] KAN-82 범용 스타일 제작

- tailwind.config.ts에 xl 반응형 크기 설정
- 프로젝트 bg-color 변경
- tilawindcss 커스텀 클래스 생성

/blog

kimgorok committed:
💄design: [블로그] KAN-82 Icon 컴포넌트 수정 및 아이콘 추가

- Icons.tsx의 fill 속성 수정
- 새로운 아이콘 추가

/blog

kimgorok committed:
Merge branch 'main' of https://github.com/KEA-5th-Myaong/frontend into KAN-82-블로그페이지레이아웃제작

kimgorok committed:
✨feat: [블로그] KAN-82 최근 포스트 목록 세팅

- 최근 포스트 목록 컴포넌트 세팅
- 포스트의 types 설정
- 테스트 json 파일 작성

/blog/[userId]

kimgorok committed:
🎉init: [블로그] KAN-82 포스트 상세 조회 페이지 세팅

- /blog/[userId]/[postTitle] 라우팅

/blog/[userId]/[postTitle]

kimgorok committed:
✨fate: [블로그] KAN-82 포스트 상세 조회 컴포넌트 세팅

- PostConents.tsx 및 PostComment.tsx 세팅
- 테스트 json 파일 작성
- 댓글 types 작성

/blog/[userId]/[postTitle]

kimgorok committed:
✨feat: [블로그] KAN-82 BackButton 컴포넌트 생성

- 'use client' 선언 위치 문제로 컴포넌트 생성

/blog/[userId]

kimgorok committed:
✨feat: [블로그] KAN-82 PostDropDown 컴포넌트 생성

- 더보기 아이콘 클릭 시 나타나는 PostDropDown 컴포넌트 생성
- 주석 추가

/blog/[userId]/[postTitle]

kimgorok committed:
🔥remove: [블로그] KAN-82 CommentInput 컴포넌트 제거

- CommentInput 컴포넌트를 PostComment 컴포넌트에 통합

/blog/[userId]/[postTitle]

kimgorok committed:
✨feat: [블로그] KAN-82 뒤로가기 아이콘 추가 및 적용

- ArrowIcon 생성 및 BackButton 컴포넌트에 적용

/blog/[userId]/[postTitle]

kimgorok committed:
✨feat: [블로그] KAN-82 layout.tsx 세팅

- /blog/[userId] 하위의 공통 layout.tsx 생성

/blog/[userId]

kimgorok committed:
✨feat: [모의면접] 아이콘 추가

- 모의면접 페이지에서 사용할 3개의 아이콘 추가

KAN-85

kimgorok committed:
🎉init: [모의면접] 페이지 세팅

- /interview/[id] 폴더구조 세팅
- 모의 면접 - 기업 선택 페이지 생성
- 면접 기록 컴포넌트 생성

KAN-85

kimgorok committed:
🛠chore: [공통] PR template 수정

- JIRA ticket 항목 추가

kimgorok commented:
모의면접 커밋 2개가 잘못들어갔어요! ㅜㅠ

kimgorok committed:
✨feat: [블로그] 팔로우/팔로잉 목록 모달 추가

- 테스트 코드 작성
- 타입 작성
- UI 적용
- 리팩토링 이전

KAN-82

kimgorok committed:
♻refactor: [블로그] 팔로우/팔로잉 리팩토링

- 팔로우 모달 작성
- 팔로우 버튼 작성
- 테스트 코드 수정

KAN-82

kimgorok committed:
Merge branch 'KAN-82-블로그페이지레이아웃제작' of https://github.com/KEA-5th-Myaong/frontend into KAN-82-블로그페이지레이아웃제작

kimgorok committed:
💄design: [블로그] 팔로잉 버튼 추가

- 임시 팔로잉 버튼 추가

KAN-82

BaekJiyeon02 commented on src/app/(route)/(blog)/blog/[userId]/_components/PostContainer.tsx:
```diff
@@ -0,0 +1,44 @@
+            postTitle={post.postTitle}
+            postContent={post.postContent}
+            postDate={post.postDate}
+            isLoved={post.isLoved}
```
이건 사용자가 포스트에 좋아요를 눌렀는지 여부를 구분하기 위한 건가요?

kimgorok replied:
네 맞습니다

BaekJiyeon02 commented on src/app/(route)/(blog)/blog/[userId]/[postTitle]/_components/PostContent.tsx:
```diff
@@ -1,17 +1,31 @@
+        <div className="cursor-pointer relative">
+          <Icons
+            onClick={() => {
+              setShowDropDown((prev) => !prev);
```
DropDown 밖의 영역을 누르면 닫히게 구현되었나요?

kimgorok replied:
아뇨 그 부분은 구현이 안되어있습니다!

kimgorok replied:
아뇨 그 부분은 구현이 안되어있습니다!

BaekJiyeon02 replied:
구현 된다면 사용자가 사용하는데 불편함이 적을 것 같아요! 🏃‍♂️

kimgorok replied:
반영하고 커밋했습니다.
공통 훅으로 useClickOutside 훅을 만들어, 다른 페이지에서도 사용할 수 있도록 하였습니다.

kimgorok committed:
✨feat: [블로그] ref 영역 밖 클릭 이벤트 훅 추가

- useClickOutside.ts 제작
- PostDropDown ref 영역 밖 클릭 시 showDropDown이 false가 되도록 적용

KAN-82

yeonilil commented on src/app/(route)/(blog)/blog/[userId]/[postTitle]/_components/BackButton.tsx:
```diff
@@ -0,0 +1,15 @@
+      뒤로가기
+    </div>
+  );
+}
```
뒤로가기 버튼 제가 디자인에 엄청 많이 넣어서 공동 Component 폴더로 옮겨도 좋을 것 같아요!

kimgorok replied:
반영했습니다!

yeonilil commented on src/app/(route)/(blog)/blog/[userId]/[postTitle]/_components/PostContent.tsx:
```diff
@@ -0,0 +1,69 @@
+  const [showDropDown, setShowDropDown] = useState(false);
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  useClickOutside({
```
모달 영역 외 처리 해주셔서 감사해요!
X 아이콘도 추가해주시면 좋을 것 같아요!

kimgorok replied:
요 훅은 더보기 목록이 나타났다 안나타났다하는 토글로 제작한거라, 이 부분에는 아이콘 추가를 안해도 될거같아요

yeonilil replied:
아하 넵넵~

yeonilil commented on src/app/(route)/(blog)/blog/[userId]/[postTitle]/_components/PostDropDown.tsx:
```diff
@@ -0,0 +1,12 @@
+export default function PostDropDown() {
+  return (
+    <div className="absolute border rounded-lg p-[10px] bg-white-0 right-0">
```
요기 드롭다운 디자인이 hover 시 초록색으로 바뀌는 걸로 구상했는데 설명을 못 적어뒀네요!!

kimgorok replied:
수정해뒀습니다

yeonilil commented on src/app/(route)/(blog)/blog/[userId]/_components/UserProfile.tsx:
```diff
@@ -0,0 +1,92 @@
+        isOpen={isFollowingOpen}
+        onClose={() => setIsFollowingOpen(false)}
+        title={`${userName}님이 팔로우하는 유저`}
+        list={followingList}
```
모달 작업까지 👍 👍

yeonilil commented on src/app/(route)/(blog)/blog/[userId]/_components/_follow/FollowModal.tsx:
```diff
@@ -0,0 +1,56 @@
+  list: FollowProps[];
+}
+
+const followMotion = {
```
애니메이션 적용해주신 부분 너무 좋아요!

yeonilil commented on src/app/(route)/(blog)/blog/[userId]/[postTitle]/_components/PostContent.tsx:
```diff
@@ -0,0 +1,69 @@
+        <span className="text-sm text-gray-0">{testPosts[0].postDate}</span>
+        <div className="flex gap-3">
+          <div className="text-primary-1 blog-favor-frame">
+            <div id="commentIcon" className="bg-primary-1 w-[18px] h-[18px] rounded-full" />
```
commentIcon이 잘 안 보이는데 혹시 확인해주실 수 있을까요? 초록색 원으로 보이네요!

kimgorok replied:
아 CommentIcon svg파일을 구하지 못해서 임시로 저렇게 해두었습니다. 나중에 수정해서 반영하겠습니다!

yeonilil replied:
아하 넵 확인했습니당

yeonilil reviewed (approved):
넘 수고 많으셨습니다 현중님! 👍 👍

kimgorok committed:
🚚rename: [블로그] 폴더 위치 변경

- BackButton을 공통 _components로 이동
- _utils 폴더를 _hooks로 변경

KAN-82

kimgorok committed:
💄design: [블로그] PostDropDown 버튼 배경 hover시 변경으로 수정

- 기존의 배경색에서, hover 시 배경색이 변경되는 방식으로 변경
- 반응형 padding 설정

KAN-82

BaekJiyeon02 reviewed (commented):
고생많으셨습니다 👏
